sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -100,3 +102,10 @@
 // clang emits "warning:" in the message pragma output
 #define __SYCL_WARNING(msg) __pragma(message(msg))
 #endif // __GNUC__
+
+// Emitting a message instead of a warning here in order to avoid hard error
```

Comment:
I typically don't negotiate with terrorists.  I suggest you don't either.

Reply:
You are saying that it should be a warning rather than a message? The only thing I'm afraid of is that if it was a warning it can break build of a lot of libraries that use SYCL. So I thought that it's better to start with a message.

Reply:
This is a bit of a 'Clark Nelson'-ism.  He believed that people who compile with -Werror are opting into warnings-causing-failures, and that we shouldn't make everyone else's behavior so much worse so that these people could work, particularly when -Wno-*** flags exist.